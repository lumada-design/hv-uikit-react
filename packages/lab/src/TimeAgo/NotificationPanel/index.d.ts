export { default } from "./NotificationPanel";
export * from "./NotificationPanel";
