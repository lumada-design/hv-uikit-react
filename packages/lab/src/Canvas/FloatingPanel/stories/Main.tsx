import { css } from "@emotion/css";
import {
  HvCanvasFloatingPanel,
  HvCanvasFloatingPanelProps,
} from "@hitachivantara/uikit-react-lab";

const tabs = [
  {
    id: 0,
    title: "Tab 1",
  },
  {
    id: 1,
    title: "Tab 2",
  },
];

const classes = { root: css({ position: "relative" }) }; // for Storybook purposes

export const MainStory = (props: HvCanvasFloatingPanelProps) => (
  <HvCanvasFloatingPanel {...props} className={classes.root} tabs={tabs}>
    Content
  </HvCanvasFloatingPanel>
);
