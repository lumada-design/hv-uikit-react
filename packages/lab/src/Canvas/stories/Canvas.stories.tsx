import { useEffect, useState } from "react";
import { css } from "@emotion/css";
import { Meta, StoryObj } from "@storybook/react";
import { Background, BackgroundVariant, MiniMap } from "reactflow";
import {
  HvButton,
  HvDropDownMenu,
  HvIconButton,
  HvInlineEditor,
  HvLoading,
  theme,
} from "@hitachivantara/uikit-react-core";
import {
  Backwards,
  Calendar,
  DataSource,
  Plane,
  Pod,
  Redo,
  Schema,
  Undo,
  User,
} from "@hitachivantara/uikit-react-icons";
import {
  HvCanvasPanel,
  HvCanvasTab,
  HvCanvasToolbar,
  HvFlow,
} from "@hitachivantara/uikit-react-lab";

import { ListView } from "./ListView/ListView";
import { eds, nds, nodeTypes } from "./nodes";
import { TreeView } from "./TreeView";

const meta: Meta = {
  title: "Lab/Canvas",
};
export default meta;

export const Main: StoryObj = {
  args: {
    title: "Semantic Model Name",
  },
  argTypes: {
    classes: { control: { disable: true } },
  },
  render: () => {
    const [tabs, setTabs] = useState<HvCanvasTab[]>([]);
    const [selectedTab, setSelectedTab] = useState<string>("1");

    const classes = {
      separator: css({
        height: 30,
        width: 1,
        backgroundColor: theme.colors.atmo4,
        margin: `0 ${theme.space.xs}`,
      }),
    };
    const backButton = (
      <HvButton aria-label="Back" icon>
        <Backwards />
      </HvButton>
    );

    const title = (
      <HvInlineEditor value="Semantic Model Name" variant="title4" />
    );

    const Separator = () => <div className={classes.separator} />;

    const renderIcon = (Icon: React.ElementType) => () => <Icon />;

    useEffect(() => {
      setTabs([
        {
          id: "1",
          content: (
            <>
              <DataSource />
              Add Data
            </>
          ),
        },
        {
          id: "2",
          content: (
            <>
              <Schema />
              Model Structure
            </>
          ),
        },
      ]);
    }, []);

    return tabs && tabs.length > 0 ? (
      <div
        style={{
          width: "100%",
          height: "100vh",
          position: "relative",
        }}
      >
        <HvFlow
          style={{ width: "100%", height: "100vh" }}
          snapGrid={[1, 1]}
          snapToGrid
          nodes={nds}
          edges={eds}
          nodeTypes={nodeTypes}
        >
          <Background
            color={theme.palette.slate[950]}
            variant={BackgroundVariant.Dots}
            gap={16}
          />
          <MiniMap />
        </HvFlow>
        <HvCanvasPanel
          tabs={tabs}
          onTabChange={(tabId) => setSelectedTab(tabId)}
        >
          {selectedTab === "1" && <TreeView />}
          {selectedTab === "2" && <ListView />}
        </HvCanvasPanel>
        <HvCanvasToolbar backButton={backButton} title={title}>
          <HvIconButton title="Undo">
            <Undo />
          </HvIconButton>
          <HvIconButton title="Redo">
            <Redo />
          </HvIconButton>
          <Separator />
          <HvIconButton title="Add">
            <Pod />
          </HvIconButton>
          <Separator />
          <HvButton variant="primary">Save</HvButton>
          <HvButton variant="primaryGhost">Cancel</HvButton>
          <Separator />
          <HvDropDownMenu
            placement="right"
            onClick={(e, item) => console.log(item.label)}
            dataList={[
              { label: "Label 1", icon: renderIcon(User) },
              { label: "Label 2", icon: renderIcon(Calendar), disabled: true },
              { label: "Label 3", icon: renderIcon(Plane) },
            ]}
          />
        </HvCanvasToolbar>
      </div>
    ) : (
      <HvLoading />
    );
  },
};
