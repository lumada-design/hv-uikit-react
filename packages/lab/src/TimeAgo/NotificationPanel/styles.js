const styles = (theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    padding: 0,
    width: 320,
    overflow: "hidden",
  },
  closed: { display: "none" },
  open: {
    position: "absolute",
    right: 0,
    boxShadow: theme.hv.shadows[1],
  },
  actionBar: {
    marginTop: "auto",
  },
  panel: {
    flexGrow: 1,
    overflow: "auto",
    padding: 0,
    width: "100%",
    "&.focus-visible": {
      outline: "none",
    },
  },

  actionBarRoot: {
    backgroundColor: theme.hv.palette.atmosphere.atmo1,
    justifyContent: "space-between",
  },

  emptyState: {
    height: "500px",
  },
});

export default styles;
