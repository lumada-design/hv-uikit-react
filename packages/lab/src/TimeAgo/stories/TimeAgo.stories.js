import React, { useMemo, useState } from "react";
import dayjs from "dayjs";

import { makeStyles } from "@material-ui/core";
import { HvRadio, HvRadioGroup, HvTypography } from "@hitachivantara/uikit-react-core";

import { HvTimeAgo, HvTimePicker } from "../..";

export default {
  title: "Lab/Time Ago",
  parameters: {
    componentSubtitle: null,
    usage: 'import { HvTimeAgo } from "@hitachivantara/uikit-react-lab"',

    dsVersion: "3.3.0",
  },
  component: HvTimeAgo,
};

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    alignItems: "center",
  },
  container: {
    minHeight: 300,
    "& > div": {
      padding: theme.hvSpacing("xs"),
    },
  },
  table: {
    border: `1px solid ${theme.palette.acce1}`,
    borderCollapse: "collapse",
    "& th, td": {
      border: `1px solid ${theme.palette.acce1}`,
      padding: theme.hvSpacing("5px", "sm"),
    },
  },
}));

export const Main = () => {
  const classes = useStyles();
  const dates = useMemo(
    () =>
      [
        dayjs(),
        dayjs().subtract(1, "minutes"),
        dayjs().subtract(10, "minutes"),
        dayjs().subtract(59, "minutes"),
        dayjs().hour(0),
        dayjs().day(0),
        dayjs().date(0),
        dayjs().month(-2),
        dayjs().month(-4),
      ].map((date) => date.valueOf()),
    []
  );

  return (
    <table className={classes.table}>
      <thead>
        <tr>
          <th>ISO Date</th>
          <th>{"<TimeAgo />"}</th>
        </tr>
      </thead>
      <tbody>
        {dates.map((dateTs) => (
          <tr key={dateTs}>
            <td>{new Date(dateTs).toISOString()}</td>
            <td>
              <HvTimeAgo timestamp={dateTs} />
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  );
};

export const LocaleOverride = () => {
  const classes = useStyles();
  const [locale, setLocale] = useState("en");
  const [time, setTime] = useState(Date.now());

  const handleTimeChange = ({ hours, minutes, seconds }) => {
    const newDate = new Date();
    newDate.setHours(hours);
    newDate.setMinutes(minutes);
    newDate.setSeconds(seconds);
    setTime(newDate.getTime());
  };

  const handleLocaleChange = async (event, newLocale) => {
    // prevent bundling of unwanted non-js files
    // by specifying the file extension in the dynamic import
    await import(`dayjs/locale/${newLocale}.js`);
    setLocale(newLocale);
  };

  return (
    <div className={classes.container}>
      <div>
        <HvRadioGroup orientation="horizontal" value={locale} onChange={handleLocaleChange}>
          <HvRadio label="ðŸ‡¬ðŸ‡§ English" value="en" />
          <HvRadio label="ðŸ‡«ðŸ‡· French" value="fr" />
          <HvRadio label="ðŸ‡©ðŸ‡ª German" value="de" />
          <HvRadio label="ðŸ‡µðŸ‡¹ Portuguese" value="pt" />
        </HvRadioGroup>
      </div>
      <div>
        <HvTypography variant="sTitle">
          <HvTimeAgo timestamp={time} locale={locale} />
        </HvTypography>
        <span>{new Date(time).toISOString()}</span>
      </div>
      <div style={{ width: 300, minHeight: 300 }}>
        <HvTimePicker label="Time" description="Pick a time" onChange={handleTimeChange} />
      </div>
    </div>
  );
};

LocaleOverride.parameters = {
  docs: {
    description: {
      story:
        // eslint-disable-next-line no-multi-str
        "Sample dates and locale controlled externally.<br /> \
        `HvTimeAgo` leverages `dayjs` for locales and custom dates. To use a locale, import it with `dayjs/locale/{locale}`<br />\
        Locale strings can be overridden using [dayjs.updateLocale](https://day.js.org/docs/en/plugin/update-locale)",
    },
  },
};
