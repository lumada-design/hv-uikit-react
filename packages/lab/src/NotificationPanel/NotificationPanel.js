import React, { Component } from "react";
import PropTypes from "prop-types";
import map from "lodash/map";
import { withStyles } from "@material-ui/core";
import { HvBadge } from "@hitachivantara/uikit-react-core";
import Panel from "./Panel";
import Notification from "./Notification";
import styles from "./styles";

/**
 * Notification Panel displays all of read and unread notifications. Still in development
 */
class HvNotificationPanel extends Component {
  constructor(props) {
    super(props);

    this.state = {
      open: props.open,
    };
  }

  handleIconClick = () => {
    this.setState((prevState) => ({
      open: !prevState.open,
    }));
  };

  handleIconKeyDown = (e) => {
    if (e.keyCode === 13) {
      this.setState((prevState) => ({
        open: !prevState.open,
      }));
    }
  };

  onClose = () => {
    this.setState({
      open: false,
    });
  };

  render() {
    const { classes, icon, header, footer, notifications } = this.props;
    const { open } = this.state;

    const n = map(notifications, (notification) => (
      <Notification key={notification.id} notification={notification} />
    ));

    const panelProps = {
      open,
      classes: {
        panel: classes.panel,
      },
      onClose: this.onClose,
      header,
      footer,
      notifications,
    };

    return (
      <div className={classes.root}>
        <div
          role="button"
          className={classes.iconWrapper}
          tabIndex="0"
          onKeyDown={this.handleIconKeyDown}
          onClick={this.handleIconClick}
        >
          <HvBadge
            icon={icon}
            showCount
            count={notifications.length}
            classes={{ root: classes.badgeBorder }}
          />
        </div>
        <Panel {...panelProps}>{n}</Panel>
      </div>
    );
  }
}

HvNotificationPanel.propTypes = {
  /**
   * 'true' if panel is open or 'false' if the panel is not open
   */
  open: PropTypes.bool,
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    root: PropTypes.string,
    iconWrapper: PropTypes.string,
    /**
     * styles object applied to the panel
     */
    panel: PropTypes.any,
    /**
     * styles object applied to badgeContainer
     */
    badgeBorder: PropTypes.any,
  }).isRequired,
  /**
   * renderable element that opens the notification panel
   */
  icon: PropTypes.element.isRequired,
  /**
   * Object that holds header properties
   */
  header: PropTypes.shape({
    /**
     * The title of the header
     */
    headerTitle: PropTypes.string.isRequired,
    /**
     * The icon that denoted close functionality
     */
    headerCloseImg: PropTypes.element.isRequired,
  }).isRequired,
  /**
   * Array of notification object to be rendered inside the panel
   */
  notifications: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * Id of notification
       */
      id: PropTypes.string.isRequired,
    })
  ).isRequired,
  /**
   * renderable footer element
   */
  footer: PropTypes.element.isRequired,
};

HvNotificationPanel.defaultProps = {
  open: false,
};

export default withStyles(styles, { name: "HvNotificationPanel" })(HvNotificationPanel);
