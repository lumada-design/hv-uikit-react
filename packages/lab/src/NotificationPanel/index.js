import withStyles from "@material-ui/core/styles/withStyles";
import styles from "./styles";
import NotificationPanel from "./NotificationPanel";

export default withStyles(styles, { name: "HvNotificationPanel" })(NotificationPanel);
