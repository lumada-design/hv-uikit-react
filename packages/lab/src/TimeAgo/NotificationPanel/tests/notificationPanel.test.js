/* eslint-env jest */
/* eslint-disable no-console */
import React from "react";

import { render } from "testing-utils";
import { Main } from "../stories/NotificationPanel.stories";

describe("Notification Panel", () => {
  describe("sample snapshot testing", () => {
    const consoleSpy = jest.fn();
    const originalError = console.error;
    beforeEach(async () => {
      consoleSpy.mockReset();
      console.error = consoleSpy;
    });

    afterEach(async () => {
      console.error = originalError;
    });
    it("Main", () => {
      const { container } = render(<Main />);
      expect(container).toMatchSnapshot();
      expect(console.error).toHaveBeenCalledTimes(1);
      expect(consoleSpy.mock.calls[0][2].includes("is deprecated")).toBe(true);
    });
  });
});
