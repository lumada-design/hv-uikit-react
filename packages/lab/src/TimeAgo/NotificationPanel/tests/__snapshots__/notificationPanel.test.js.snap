// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification Panel sample snapshot testing Main 1`] = `
<div>
  <div
    style="width: 100%;"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root HvHeader-root MuiAppBar-colorPrimary HvHeader-backgroundColor MuiAppBar-positionRelative css-1j4xqc3-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="HvHeader-header"
      >
        <nav
          class="HvHeaderNavigation-root"
        >
          <div
            class="HvHeaderMenuBar-root HvHeaderMenuBar-menubar"
          >
            <ul
              class="HvHeaderMenuBar-list"
            >
              <li
                class="HvHeaderMenuItem-root HvHeaderMenuItem-notSelectedItem"
              >
                <div
                  class="HvHeaderMenuItem-button"
                  role="button"
                  tabindex="0"
                >
                  <p
                    class="HvTypography-root HvTypography-normalText"
                  >
                    Overview
                  </p>
                </div>
              </li>
              <li
                class="HvHeaderMenuItem-root HvHeaderMenuItem-notSelectedItem"
              >
                <div
                  class="HvHeaderMenuItem-button"
                  role="button"
                  tabindex="0"
                >
                  <p
                    class="HvTypography-root HvTypography-normalText"
                  >
                    Events
                  </p>
                </div>
              </li>
              <li
                class="HvHeaderMenuItem-root HvHeaderMenuItem-notSelectedItem"
              >
                <div
                  class="HvHeaderMenuItem-button"
                  role="button"
                  tabindex="0"
                >
                  <p
                    class="HvTypography-root HvTypography-normalText"
                  >
                    Asset
                  </p>
                </div>
              </li>
            </ul>
          </div>
        </nav>
        <div
          class="HvHeaderActions-root"
        >
          <button
            aria-label="Open Notifications panel"
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium makeStyles-panelOpen-1 HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <div
              class="HvBadge-root"
            >
              <div
                class="HvIconBase-root HvIconBase-s"
                name="Alert"
              >
                <svg
                  focusable="false"
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M14 12V8a5.884 5.884 0 00-4-5.666V2a2 2 0 00-4 0v.334A5.884 5.884 0 002 8v4H0v1h6v1a2 2 0 004 0v-1h6v-1zM7 2a1 1 0 012 0v.086a5.85679 5.85679 0 00-2 0zm2 12a1 1 0 01-2 0v-1h2zm-6-2V8a5 5 0 0110 0v4z"
                    fill="#414141"
                  />
                </svg>
              </div>
              <div
                class="HvBadge-badgeContainer"
              >
                <div
                  class="HvBadge-badge HvBadge-badgeIcon"
                />
              </div>
            </div>
          </button>
          <button
            aria-label="Open User panel"
            class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <div
              class="HvIconBase-root HvIconBase-s"
              name="User"
            >
              <svg
                focusable="false"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="color0"
                  d="M9.6 8.7A4.49656 4.49656 0 008.03809 0Q8.019-.00007 8 0a4.481 4.481 0 00-4.5 4.46194q-.00009.019 0 .03808A4.563 4.563 0 006.4 8.7 7.687 7.687 0 00.5 16h1a6.487 6.487 0 016.474-6.5H8a6.487 6.487 0 016.5 6.474V16h1a7.493 7.493 0 00-5.9-7.3zM4.5 4.5A3.543 3.543 0 018 1a3.543 3.543 0 013.5 3.5A3.543 3.543 0 018 8a3.543 3.543 0 01-3.5-3.5z"
                  fill="#414141"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </header>
    <div
      class="HvNotificationPanel-root HvNotificationPanel-open HvPanel-root MuiBox-root css-0"
      style="right: 30px; height: calc(100% - 130px);"
    >
      <div
        class="HvNotificationPanel-panel"
      >
        <div
          class="HvNotificationPanelNotification-root"
          role="button"
          tabindex="0"
        >
          <div
            class="HvNotificationPanelNotification-notificationWrapper HvNotificationPanelNotification-clickable"
          >
            <div
              class="HvNotificationPanelNotification-iconContainer"
            >
              <div
                class="HvIconBase-root HvIconBase-s"
                name="Level5"
              >
                <svg
                  focusable="false"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M12 0A12 12 0 110 12 12 12 0 0112 0z"
                    fill="#C62828"
                  />
                  <path
                    class="color1"
                    d="M10.88 15.81h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm4 9.87h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm-8 9.87h2.25v2.25H6.88zm0-9.87h2.25v8H6.88z"
                    fill="#F0F0F0"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                class=""
                data-mui-internal-clone-element="true"
              >
                <div
                  class="HvNotificationPanelNotification-messageContainer"
                >
                  <p
                    class="HvTypography-root HvTypography-highlightText"
                  >
                    Cabin door knob malfunction. Super long test title, to test line break behavior.
                  </p>
                </div>
              </div>
              <div
                class="HvNotificationPanelNotification-timeContainer"
              >
                <div
                  class="HvNotificationPanelNotification-bullet"
                />
                <div>
                  a few seconds ago
                </div>
              </div>
            </div>
          </div>
          <div
            class="HvNotificationPanelNotification-notificationActionWrapper"
          >
            <div
              class="HvBaseDropdown-root HvDropDownMenu-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="menu"
                class="HvDropDownMenu-container HvBaseDropdown-anchor"
                id="drop-down-menu-1"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-label="Dropdown menu"
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvDropDownMenu-icon HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
                  id="drop-down-menu-1-icon-button"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="HvIconBase-root HvIconBase-s"
                    name="MoreOptionsVertical"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="color0"
                        d="M7 6.9h2v2H7zM7 4h2V2H7zm0 10h2v-2H7z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="HvNotificationPanelNotification-root"
          role="button"
          tabindex="0"
        >
          <div
            class="HvNotificationPanelNotification-notificationWrapper HvNotificationPanelNotification-clickable"
          >
            <div
              class="HvNotificationPanelNotification-iconContainer"
            />
            <div>
              <div
                class=""
                data-mui-internal-clone-element="true"
              >
                <div
                  class="HvNotificationPanelNotification-messageContainer"
                >
                  <p
                    class="HvTypography-root HvTypography-highlightText"
                  >
                    Cabin door knob malfunction to second line
                  </p>
                </div>
              </div>
              <div
                class="HvNotificationPanelNotification-timeContainer"
              >
                <div
                  class="HvNotificationPanelNotification-bullet"
                />
                <div>
                  a few seconds ago
                </div>
              </div>
            </div>
          </div>
          <div
            class="HvNotificationPanelNotification-notificationActionWrapper"
          >
            <div
              class="HvBaseDropdown-root HvDropDownMenu-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="menu"
                class="HvDropDownMenu-container HvBaseDropdown-anchor"
                id="drop-down-menu-2"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-label="Dropdown menu"
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvDropDownMenu-icon HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
                  id="drop-down-menu-2-icon-button"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="HvIconBase-root HvIconBase-s"
                    name="MoreOptionsVertical"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="color0"
                        d="M7 6.9h2v2H7zM7 4h2V2H7zm0 10h2v-2H7z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="HvNotificationPanelNotification-root HvNotificationPanelNotification-read"
          role="button"
          tabindex="0"
        >
          <div
            class="HvNotificationPanelNotification-notificationWrapper HvNotificationPanelNotification-clickable"
          >
            <div
              class="HvNotificationPanelNotification-iconContainer"
            >
              <div
                class="HvIconBase-root HvIconBase-s"
                name="Level5"
              >
                <svg
                  focusable="false"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M12 0A12 12 0 110 12 12 12 0 0112 0z"
                    fill="#C62828"
                  />
                  <path
                    class="color1"
                    d="M10.88 15.81h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm4 9.87h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm-8 9.87h2.25v2.25H6.88zm0-9.87h2.25v8H6.88z"
                    fill="#F0F0F0"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                class=""
                data-mui-internal-clone-element="true"
              >
                <div
                  class="HvNotificationPanelNotification-messageContainer"
                >
                  <p
                    class="HvTypography-root HvTypography-normalText"
                  >
                    Cabin door knob malfunction
                  </p>
                </div>
              </div>
              <div
                class="HvNotificationPanelNotification-timeContainer"
              >
                <div
                  class="HvNotificationPanelNotification-bullet HvNotificationPanelNotification-hide"
                />
                <div>
                  19 June 2019, 12:00 AM
                </div>
              </div>
            </div>
          </div>
          <div
            class="HvNotificationPanelNotification-notificationActionWrapper"
          >
            <div
              class="HvBaseDropdown-root HvDropDownMenu-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="menu"
                class="HvDropDownMenu-container HvBaseDropdown-anchor"
                id="drop-down-menu-3"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-label="Dropdown menu"
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvDropDownMenu-icon HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
                  id="drop-down-menu-3-icon-button"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="HvIconBase-root HvIconBase-s"
                    name="MoreOptionsVertical"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="color0"
                        d="M7 6.9h2v2H7zM7 4h2V2H7zm0 10h2v-2H7z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="HvNotificationPanelNotification-root HvNotificationPanelNotification-read"
          role="button"
          tabindex="0"
        >
          <div
            class="HvNotificationPanelNotification-notificationWrapper"
          >
            <div
              class="HvNotificationPanelNotification-iconContainer"
            >
              <div
                class="HvIconBase-root HvIconBase-s"
                name="Level5"
              >
                <svg
                  focusable="false"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    class="color0"
                    d="M12 0A12 12 0 110 12 12 12 0 0112 0z"
                    fill="#C62828"
                  />
                  <path
                    class="color1"
                    d="M10.88 15.81h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm4 9.87h2.25v2.25h-2.25zm0-9.87h2.25v8h-2.25zm-8 9.87h2.25v2.25H6.88zm0-9.87h2.25v8H6.88z"
                    fill="#F0F0F0"
                  />
                </svg>
              </div>
            </div>
            <div>
              <div
                class=""
                data-mui-internal-clone-element="true"
              >
                <div
                  class="HvNotificationPanelNotification-messageContainer"
                >
                  <p
                    class="HvTypography-root HvTypography-normalText"
                  >
                    Cabin door knob malfunction
                  </p>
                </div>
              </div>
              <div
                class="HvNotificationPanelNotification-timeContainer"
              >
                <div
                  class="HvNotificationPanelNotification-bullet HvNotificationPanelNotification-hide"
                />
                <div>
                  19 June 2019, 12:00 AM
                </div>
              </div>
            </div>
          </div>
          <div
            class="HvNotificationPanelNotification-notificationActionWrapper"
          >
            <div
              class="HvBaseDropdown-root HvDropDownMenu-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="menu"
                class="HvDropDownMenu-container HvBaseDropdown-anchor"
                id="drop-down-menu-4"
                tabindex="-1"
              >
                <button
                  aria-expanded="false"
                  aria-label="Dropdown menu"
                  class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvDropDownMenu-icon HvButton-icon HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
                  id="drop-down-menu-4-icon-button"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="HvIconBase-root HvIconBase-s"
                    name="MoreOptionsVertical"
                  >
                    <svg
                      focusable="false"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="color0"
                        d="M7 6.9h2v2H7zM7 4h2V2H7zm0 10h2v-2H7z"
                        fill="#414141"
                      />
                    </svg>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="HvActionBar-root HvNotificationPanel-actionBarRoot HvNotificationPanel-actionBar"
      >
        <button
          class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Mark All Read
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root HvButton-root MuiButton-text MuiButton-textPrimary HvButton-ghost MuiButton-sizeMedium MuiButton-textSizeMedium HvButton-ghostSVG css-fsero8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Settings
        </button>
      </div>
    </div>
  </div>
</div>
`;
